block/manager.go

```diff
@@ -1210,11 +1210,18 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 
 		batchData, err := m.retrieveBatch(ctx)
 		if errors.Is(err, ErrNoBatch) {
-			return fmt.Errorf("no batch to process")
+			m.logger.Info("No batch retrieved from sequencer, skipping block production")
+			return nil // Indicate no block was produced
 		} else if err != nil {
 			return fmt.Errorf("failed to get transactions from batch: %w", err)
 		}
 
+		// Check if the batch contains any transactions
+		if len(batchData.Batch.Transactions) == 0 {
```

Comment:
we could move this check inside retrieveBatch itself and return ErrNoBatch?

Reply:
moved, want to double check its what you meant